File: src/main/java/com/fengsheng/handler/po_yi_show_tos.java
package com.fengsheng.handler;

import com.fengsheng.HumanPlayer;
import com.fengsheng.protos.Fengsheng;
import org.apache.log4j.Logger;

public class po_yi_show_tos extends AbstractProtoHandler<Fengsheng.po_yi_show_tos> {
    private static final Logger log = Logger.getLogger(po_yi_show_tos.class);

    @Override
    protected void handle0(HumanPlayer r, Fengsheng.po_yi_show_tos pb) {
        if (!r.checkSeq(pb.getSeq())) {
            log.error("操作太晚了, required Seq: " + r.getSeq() + ", actual Seq: " + pb.getSeq());
            return;
        }
        r.getGame().tryContinueResolveProtocol(r, pb);
    }
}

File: src/main/kotlin/handler/po_yi_show_tos.kt
package com.fengsheng.handler

import com.fengsheng.HumanPlayer
import com.fengsheng.protos.Fengsheng

import org.apache.logging.log4j.kotlin.logger

class po_yi_show_tos : AbstractProtoHandler<Fengsheng.po_yi_show_tos>() {
    override fun handle0(r: HumanPlayer, pb: Fengsheng.po_yi_show_tos) {
        if (!r.checkSeq(pb.seq)) {
            logger.error("操作太晚了, required Seq: ${r.seq}, actual Seq: ${pb.seq}")
            r.sendErrorMessage("操作太晚了")
            return
        }
        r.game!!.tryContinueResolveProtocol(r, pb)
    }

    companion object {
    }
}